web/src/recent_view_data.ts

```diff
@@ -48,13 +48,24 @@ export function process_message(msg: Message): boolean {
     return conversation_data_updated;
 }
 
-function get_sorted_topics(): Map<string | undefined, TopicData> {
+function get_sorted_topics(topic_map: Map<string, TopicData>): Map<string | undefined, TopicData> {
     // Sort all recent topics by last message time.
-    return new Map([...topics.entries()].sort((a, b) => b[1].last_msg_id - a[1].last_msg_id));
+    return new Map([...topic_map.entries()].sort((a, b) => b[1].last_msg_id - a[1].last_msg_id));
 }
 
 export function get(): Map<string | undefined, TopicData> {
-    return get_sorted_topics();
+    return get_sorted_topics(topics);
+}
+
+export function get_topics_for_stream_id(stream_id: number): Map<string | undefined, TopicData> {
+    const stream_topics = new Map(
+        [...topics.entries()].filter(
+            (topic) =>
+                // TODO: It would be great if the map had the stream_id directly accessible.
+                topic[1].type === "stream" && topic[0].startsWith(`${stream_id.toString()}:`),
```

Comment:
I think this might not be a refactor worth investing in at this time; the fairly near-term plan should be to have recent topics data have its own independent data structure based on a  `MessageListData` object for the stream that we fetch data for independently, that's initialized from `all_messages_data` (just like the ones for narrows are), that has the same type as the main "recent topics" data structure.

(One possibility is that we change this module into a class that can be attached to any `MessageListData` object that is the "topics summary" of that sequence of messages).

Reply:
Oh yeah I meant for this TODO to be a "it would be cool but out of scope for this PR" -- is there a better way to mark that?